Class.getDeclaredFields throws exception after invoking Class.newInstance

Accordingly to Javadoc of Class.getDeclaredFields, "...an array of Field objects reflecting all the fields declared by the class or interface represented by this Class object" must be returned. But using the following program declared fields of class B can not be retrieved:

```
public class A {
   static {
      String x = null;
      x.getClass();
   }
}

public class B extends A {
   public int x;
}

public class C {
   public static void main(String[] args) throws NoSuchMethodException, NoSuchFieldException {
      try {
         B.class.newInstance();
      } catch (ExceptionInInitializerError e) {
      } catch (Exception e) {
      } finally {
         for (Field f : B.class.getDeclaredFields()) {
            System.out.println(f);
         }
      }
   }
}
```

**Affected versions:**

openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-b12)
Eclipse OpenJ9 VM (build openj9-0.8.0, JRE 1.8.0 Linux amd64-64 Compressed References 20180315_120 (JIT enabled, AOT enabled)
OpenJ9   - e24e8aa9
OMR      - 3e8296b4
JCL      - ee1e77df1d based on jdk8u162-b12)

Steps to reproduce:

	1. Compile above program
	2. Run it using openj9 java

**Current result:**

Exception in thread "main" java.lang.NoClassDefFoundError: samples.newInstance.B (initialization failure)
	at java.lang.J9VMInternals.initializationAlreadyFailed(J9VMInternals.java:96)
	at java.lang.J9VMInternals.prepareClassImpl(Native Method)
	at java.lang.J9VMInternals.prepare(J9VMInternals.java:300)
	at java.lang.Class.getDeclaredFields(Class.java:830)
	at samples.newInstance.C.main(C.java:27)
Caused by: java.lang.NullPointerException
	at samples.newInstance.A.<clinit>(A.java:7)
	at java.lang.J9VMInternals.newInstanceImpl(Native Method)
	at java.lang.Class.newInstance(Class.java:1773)
	at samples.newInstance.C.main(C.java:11)

**Expected result:**

public int x

--- a/runtime/vm/ClassInitialization.cpp
+++ b/runtime/vm/ClassInitialization.cpp
@@ -464,7 +464,14 @@ doVerify:
 				break;
 			}
 			case J9ClassInitFailed: {
-				sendInitializationAlreadyFailed(currentThread, clazz, 0, 0, 0);
+				/* J9ClassInitFailed can only be set when unlockedStatus is J9ClassInitNotInitialized, and desiredState is J9_CLASS_INIT_INITIALIZED.
+				 * J9ClassInitFailed can be ignored if desiredState is not J9_CLASS_INIT_INITIALIZED, i.e., J9_CLASS_INIT_VERIFIED or J9_CLASS_INIT_PREPARED.
+				 */
+				if (desiredState < J9_CLASS_INIT_INITIALIZED) {
+					Trc_VM_classInitStateMachine_desiredStateReached(currentThread);
+				} else {
+					sendInitializationAlreadyFailed(currentThread, clazz, 0, 0, 0);
+				}
 				goto done;
 			}
 			case J9ClassInitNotInitialized: {

